### DISCUSSION TITLE
Add new layer types

### DISCUSSION THREAD

**Discussion by lepigocher** [maplibre/maplibre#174] [Cross-project feature proposals]
It could be really usefull to have some other "geometrics" layer type like : square, triangle, hexagon, pentagon and so on. The circle type is not enougth for complex maps with many data types. In fact, we have to use png images with the "symbol" type and this is times consuming.

**Comment by wipfli** [comment]
Thanks for this suggestion. I think it would be best to open an issue in MapLibre GL JS and Native if there is not already one for this issue.

**Comment by ghost** [comment]
I feel this is too easy to handle with a symbol-layer and an image for each of those basic shapes (ideally an SDF).
Quality wise it will be similar, performance will be slightly worse, but you gain a lot of flexibility.

One extra benefit of the symbol-layer is that you can have different shapes on the same layer (which is good for performance and ability to sort items).
The icon-halo should be usable for the stroke on SDFs.

> In fact, we have to use png images with the "symbol" type and this is times consuming.

Can you explain why this is "time consuming"?

Creation of the shape images should be trivial and you can easily find images of those shapes online (although, potentially not the SDF directly).
You can also easily [generate those SDF; the formulas are readily available](https://iquilezles.org/articles/distfunctions2d/) if you use code to create your map.

If you use a style editor, I feel this icon-generation or "high-level-layer" should be the responsibility of the style editor.
(Similar to how they should generate mapbox expressions for you)

**Reply by lepigocher** [reply]
@JannikGM Sorry for the delay, I missed the notification! I already do it with png images as symbols. It's time consuming because for my use case, each project can have until hundred images used by severals thousands of 'points' features with sometimes differents rendering rules. As a sample, an equipment (rendered as an image) can be 'opened', 'closed', 'abandonned' and so on and of course, each state need the same symbol with a different color! Create all variants is really time consuming and boring, believe me! :)

I have not tried SDF icons because I got source images as SVG and even if I could save time because the color can be managed with SDF icon, I have to convert all of those icons as SDF font and I do not have found a really convenient tool to do it.

Thanks for the link about the SDF article, I will have a look on it! :)

**Reply by HarelM** [reply]
You can use spritezero, I think i managed to use it thorough docker:
https://www.npmjs.com/package/@elastic/spritezero
This is what I did in order to convert svg to sdf.

**Reply by lepigocher** [reply]
@HarelM Thanks, I will have a look at this tool, may be it could help!
